app/models/notification.rb

```diff
@@ -41,9 +41,17 @@ class Notification < ApplicationRecord
   validates :account_id, uniqueness: { scope: [:activity_type, :activity_id] }
   validates :activity_type, inclusion: { in: TYPE_CLASS_MAP.values }
 
-  scope :browserable, ->(exclude_types = []) {
+  scope :browserable, ->(exclude_types = [], from_account = nil) {
     types = TYPE_CLASS_MAP.values - activity_types_from_types(exclude_types + [:follow_request])
-    where(activity_type: types)
+    return where(activity_type: types) if from_account.nil?
+
+    username, domain = from_account.split('@')
+    account = if TagManager.instance.local_domain?(domain)
```

Comment:
I think you should definitely not reference the account by username/domain here and instead use the ID, that would simplify this scope significantly.

Reply:
done